drivers/virtio/virtio-serial.c

```diff
@@ -558,6 +562,16 @@ static int virtio_serial_probe(FAR struct virtio_device *vdev)
       goto err_with_init;
     }
 
+#ifdef CONFIG_DRIVERS_VIRTIO_SERIAL_CONSOLE
+  if (g_virtio_console == NULL)
+    {
+      DEBUGVERIFY(uart_register("/dev/console", &priv->udev));
```

Comment:
it's better to drop this change and utilize this general approach: https://github.com/apache/nuttx/pull/12764